kolibri/auth/permissions/auth.py

```diff
@@ -125,3 +127,59 @@ def readable_by_user_filter(self, user, queryset):
             return queryset.filter(id=user.dataset_id)
         else:
             return queryset.none()
+
+
+class CoachesCanManageGroupsForTheirClasses(BasePermissions):
+    def _user_is_coach_for_classroom(self, user, obj):
+        # make sure the target object is a group and user is a coach for the group's classroom
+        return obj.kind == LEARNERGROUP and user.has_role_for_collection(COACH, obj.parent)
+
+    def user_can_create_object(self, user, obj):
+        return self._user_is_coach_for_classroom(user, obj)
+
+    def user_can_read_object(self, user, obj):
+        return self._user_is_coach_for_classroom(user, obj)
```

Comment:
As mentioned on Slack, this can be `return False`, as this permission is already granted to coaches separately (and this class is about "managing" groups).